### DISCUSSION TITLE
Trying to hard delete from Purges folder does nothing

### DISCUSSION THREAD

**Discussion by joker197cinque** [microsoft/mfcmapi#564] [General]
Hi, thanks for the fantastic tool. Is it so powerful, really.

I am trying to delete some data from a mailbox and I am not able to empty the "RecoverableItems\Purges" folder. When I try to delete (using hard delete) I click OK and it does simply nothing.
I tried to enable debug log but nothing is written into it, to maybe help you for analysis.

Please help

**Comment by joker197cinque** [comment]
Edit: if I try to empty Purges folder using right click -> Advanced --> Empty items and subfolders from folder, after a while I get this error:

Warning: 
Code: MAPI_W_PARTIAL_COMPLETION == 0x00040680
Function lpMAPIFolderToEmpty->EmptyFolder( lpProgress ? reinterpret_cast<ULONG_PTR>(m_hWnd) : NULL, lpProgress, ulFlags)
File D:\a\1\s\UI\Dialogs\HierarchyTable\MsgStoreDlg.cpp
Line 717

**Reply by bradhugh** [reply]
If this is an Exchange mailbox, it's probably because SingleItemRecovery is enabled. If SIR is enabled, items can't be deleted permanently until they have reached the value in RetainDeletedItemsFor setting. See https://learn.microsoft.com/en-us/exchange/recipients-in-exchange-online/manage-user-mailboxes/enable-or-disable-single-item-recovery

**Reply by joker197cinque** [reply]
Actually I have it disabled since more than 48 hours. Also RetainDeletedItemsFor  is set to 0. I am out of luck then ?

**Reply by bradhugh** [reply]
Other than that, there might be some kind of retention policies or holds applied to the mailbox. But I'm not an expert in that area, so I wouldn't know what to tell you to check regarding that.

**Comment by joker197cinque** [comment]
Same for "RecoverableItems\Audit" folder

**Reply by bradhugh** [reply]
Audits are intentionally not able to be deleted by the end-user. This would give an attacker the ability to cover his or her tracks.

**Reply by joker197cinque** [reply]
I was able in the past to empty this folder but I can't remember how. Maybe I didn't use this tool ... few years have passed so I dont' know :(

**Comment by Snoopy5000** [comment]
Go to your EAC Exchange admin center --> Open users mailbox settings --> Click Mailbox Tab --> Manage Mailbox Policies -->Set Retention policy to [No Policy].

Then open a second powershell.  different from the one you are running the Get-MailboxFolderStatistics command from. 
#Connects you to exchange online (You must be a global admin) Copy and paste below:
Connect-ExchangeOnline

#Loop Folder Assistant Until Data is cleared. Copy and paste below and let it run side-by-side mfcmapi:
while($true) 
    {
        Start-ManagedFolderAssistant -Identity username -holdcleanup
        write-host "waiting"; start-sleep -seconds 30;
    }  

Be sure to connect Outlook to the account you want to clean and turn off cache mode, then close and reopen outlook so it takes effect.  Close Outlook.

After doing the above, the error in this topic went away, and I could see the mailbox reduce size.

**Comment by joker197cinque** [comment]
Than you so much, I will try it.

Il gio 4 mag 2023, 19:28 tarnold000 ***@***.***> ha scritto:
